frontend/www/js/omegaup/components/user/MainInfo.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div class="card">
+    <omegaup-countryflag
+      v-if="profile.country_id"
+      class="m-1"
+      :country="profile.country_id"
+    />
+    <div class="card-body">
+      <div class="img-thumbnail rounded-circle bottom-margin">
+        <img class="rounded-circle" :src="profile.gravatar_92" />
+      </div>
+    </div>
+    <div class="card-title text-center">
+      <div class="mb-3">
+        <omegaup-user-username
+          :classname="profile.classname"
+          :username="profile.username"
+        ></omegaup-user-username>
+      </div>
+      <div class="mb-3">
+        <h4 v-if="profile.rankinfo.rank > 0" class="m-0">
+          {{ `#${profile.rankinfo.rank}` }}
+        </h4>
+        <small v-else>
+          <strong> {{ rank }} </strong>
+        </small>
+        <p>
+          <small>
+            {{ T.profileRank }}
+          </small>
+        </p>
+      </div>
+      <div v-if="profile.is_own_profile || !profile.is_private" class="mb-3">
+        <h4 class="m-0">
+          {{ Object.keys(solvedProblems).length }}
+        </h4>
+        <p>
+          <small>{{ T.profileSolvedProblems }}</small>
+        </p>
+      </div>
+      <div
+        v-if="
+          profile.preferred_language &&
+          (profile.is_own_profile || !profile.is_private)
+        "
+        class="mb-3"
+      >
+        <h5 class="m-0">
+          {{
+            profile.programming_languages[profile.preferred_language].split(
+              ' ',
+            )[0]
+          }}
+        </h5>
+        <p>
+          <small>{{ T.userEditPreferredProgrammingLanguage }}</small>
+        </p>
+      </div>
+    </div>
+    <div v-if="profile.is_own_profile" class="mb-3 text-center">
+      <a class="btn btn-primary btn-sm" :href="buttonUrl">{{ buttonText }}</a>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import T from '../../lang';
+import country_Flag from '../CountryFlag.vue';
+import user_Username from './Username.vue';
+import { types } from '../../api_types';
+import { Problem } from '../../linkable_resource';
+
+@Component({
+  components: {
+    'omegaup-countryflag': country_Flag,
+    'omegaup-user-username': user_Username,
+  },
+})
+export default class UserProfileEdit extends Vue {
```

Comment:
Que te parece si renombramos este componente a:
```suggestion
export default class UserMainInfo extends Vue {
```
? para que indique lo que realmente muestra.